e2e/fixtures/fixtures.go

```diff
@@ -91,7 +91,7 @@ func NewGoDeployment(t testing.TB, d DeploymentType) *appsv1.Deployment {
 				Spec: corev1.PodSpec{
 					Containers: []corev1.Container{{
 						Name:  "container",
-						Image: "mcr.microsoft.com/oss/go/microsoft/golang:1.18",
```

Comment:
I tried using 1.20 for client/server in RP but the linter threw an error with it saying some bazel thing was incompatible beyond 1.19, so for the sake of having the same version as RP in our e2e tests maybe use 1.19 instead?

Reply:
We don't need to have the same Go version as RP

Reply:
I don't think we should hurt the quality of testing in this repo because of a limitation in RP. Being on a later version is better here. RP will eventually have to update to match.

If our goal is to completely replicate the tests here inside RP we should import them from here instead of copy / pasting them. That's not our goal though. Our goal is to have e2e tests here that ensure the operator works as expected and in rp the e2e tests ensure that the entire process works as expected.